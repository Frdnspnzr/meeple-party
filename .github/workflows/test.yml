name: Test
on:
  pull_request:
    branches:
      - "*"
  workflow_dispatch:
    inputs:
      update-snapshots:
        description: "Update snapshots?"
        type: boolean
permissions:
  contents: read
  pages: write
  id-token: write
jobs:
  build:
    timeout-minutes: 10
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v4
      - uses: actions/setup-node@v4
        with:
          node-version: lts/*
          cache: "npm"
      - name: Install dependencies
        run: npm install
      - name: Generate database bindings
        run: npm run "prisma generate"
      - name: Build app
        run: npm run build
      - name: Save build
        uses: actions/upload-artifact@v4
        with:
          name: build
          path: .next

  e2e:
    timeout-minutes: 60
    runs-on: ubuntu-latest
    needs: build
    services:
      postgres:
        image: postgres
        env:
          POSTGRES_PASSWORD: postgres
        options: >-
          --health-cmd pg_isready
          --health-interval 10s
          --health-timeout 5s
          --health-retries 5
        ports:
          - 5432:5432
    steps:
      - uses: actions/checkout@v4
      - uses: actions/setup-node@v4
        with:
          node-version: lts/*
          cache: "npm"
      - name: Install dependencies
        run: npm install
      - name: Install Playwright browsers
        run: npx playwright install --with-deps
      - name: Load build
        uses: actions/download-artifact@v4
        with:
          name: build
          path: .next
      - name: Set up snapshot cache
        id: snapshots_cache
        uses: actions/cache@v4
        with:
          key: ${{ runner.os }}-snapshots-${{github.ref}}
          restore-keys: ${{ runner.os }}-snapshots-refs/heads/main
          path: ./tests/e2e/snapshots/**
      - name: Update snapshots
        if: ${{steps.snapshots_cache.outputs.cache-hit != 'true' || inputs.update-snapshots == 'true'}}
        run: npx playwright test --update-snapshots
      - name: Generate database bindings
        run: npm run "prisma generate"
      - name: Generate database tables
        run: npm run "prisma deploy"
        env:
          DATABASE_URL: postgresql://postgres:postgres@localhost:5432/test
      - name: Run Playwright tests
        run: npx playwright test --max-failures=10
        env:
          DATABASE_URL: postgresql://postgres:postgres@localhost:5432/test
          NEXT_PUBLIC_SECRET: Testrunner
          NEXTAUTH_URL: http://127.0.0.1:3000
      - uses: actions/upload-artifact@v4
        if: always()
        with:
          name: playwright-report
          path: playwright-report/

  unittest:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v3
      - name: Setup Node.js
        uses: actions/setup-node@v3
        with:
          node-version: lts/*
          cache: "npm"
      - name: Install dependencies
        run: npm install
      - name: Generate database bindings
        run: npm run "prisma generate"
      - name: Run the tests
        run: npm test
      - name: Upload coverage reports to Codecov
        uses: codecov/codecov-action@v3
        env:
          CODECOV_TOKEN: ${{ secrets.CODECOV_TOKEN }}

  publish-e2e:
    runs-on: ubuntu-latest
    needs: [e2e, build]
    if: ${{ !cancelled() && (needs.build.result=='success') }}
    steps:
      - name: Configure GitHub Pages
        uses: actions/configure-pages@v3
      - name: Load report
        uses: actions/download-artifact@v4
        with:
          name: playwright-report
          path: playwright-report/
      - name: Upload Pages Artifact
        uses: actions/upload-pages-artifact@v2
        with:
          path: playwright-report/
      - name: Deploy to GitHub Pages
        uses: actions/deploy-pages@v2
