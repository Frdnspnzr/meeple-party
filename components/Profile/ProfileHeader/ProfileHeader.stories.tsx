import type { Meta, StoryObj } from "@storybook/react";
import ProfileHeader from "./ProfileHeader";

const meta = {
  title: "Structures/Profile header",
  component: ProfileHeader,
  tags: ["autodocs"],
} satisfies Meta<typeof ProfileHeader>;

export default meta;
type Story = StoryObj<typeof meta>;

export const FullUserDetails: Story = {
  args: {
    user: {
      name: "Frdnspnzr",
      realName: "Pascal Greilach",
      bggName: "Friedenspanzer",
      about: "Help, this thing is keeping me from playing games.",
      id: "abc123",
      place: "Kaiserslautern",
      role: "ADMIN",
      image: null,
    },
  },
};

export const MinimalUserDetails: Story = {
  args: {
    user: {
      name: "Frdnspnzr",
      realName: null,
      bggName: null,
      about: null,
      id: "abc123",
      place: null,
      role: "USER",
      image: null,
    },
  },
};

export const MyOwnProfile: Story = {
  args: {
    user: {
      name: "Frdnspnzr",
      realName: "Pascal Greilach",
      bggName: "Friedenspanzer",
      about: "Help, this thing is keeping me from playing games.",
      id: "abc123",
      place: "Kaiserslautern",
      role: "FRIENDS_FAMILY",
      image: null,
    },
    myself: true,
  },
};

export const FriendProfile: Story = {
  args: {
    user: {
      name: "Frdnspnzr",
      realName: "Pascal Greilach",
      bggName: "Friedenspanzer",
      about: "Help, this thing is keeping me from playing games.",
      id: "abc123",
      place: "Kaiserslautern",
      role: "PREMIUM",
      image: null,
    },
    friend: true,
  },
};
