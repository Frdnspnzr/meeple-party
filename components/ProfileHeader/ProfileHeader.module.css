.name {
  font-weight: 500;
}

.realName {
  font-weight: 200;
}

.subline {
  font-weight: 200;
}