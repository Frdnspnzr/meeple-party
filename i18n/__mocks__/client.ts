
export function useTranslation(ns?: string, options?: any){
  return {
    ready: true,
    t: (key: string) => key,
    i18n: null
  }
}