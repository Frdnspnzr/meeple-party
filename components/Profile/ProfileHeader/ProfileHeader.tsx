"use client";

import FriendRequestStatus from "@/components/FriendRequests/FriendRequestStatus/FriendRequestStatus";
import { UserProfile } from "@/datatypes/userProfile";
import useLayout from "@/hooks/useLayout";
import { useTranslation } from "@/i18n/client";
import { Group, MantineTheme, Stack } from "@mantine/core";
import { Role } from "@prisma/client";
import Avatar from "../../Avatar/Avatar";
import LinkButton from "../../LinkButton/LinkButton";
import { ProfileAdditionalInformation } from "../AdditionalInformation/ProfileAdditionalInformation";
import ProfileBadge from "../ProfileBadge/ProfileBadge";
import ProfileRealName from "../RealName/ProfileRealName";
import ShareProfile from "../ShareProfile/ShareProfile";
import ProfileUsername from "../Username/ProfileUsername";

export interface ProfileHeaderProps {
  user: UserProfile;
  myself?: boolean;
  friend?: boolean;
}

export default function ProfileHeader({
  user,
  myself = false,
  friend = false,
}: Readonly<ProfileHeaderProps>) {
  const { isMobile } = useLayout();
  const containerStyle = (theme: MantineTheme) => ({
    backgroundColor: theme.colors[theme.primaryColor][2],
  });
  if (isMobile) {
    return (
      <Stack align="center" style={containerStyle} p="md">
        <AvatarBlock user={user} myself={myself} friend={friend} />
        <NameBlock user={user} />
        <ButtonBlock user={user} myself={myself} />
      </Stack>
    );
  } else {
    return (
      <Group justify="space-between" style={containerStyle} p="md">
        <Group gap="xl">
          <Stack gap="xs" align="center">
            <AvatarBlock user={user} myself={myself} friend={friend} />
          </Stack>
          <Stack gap="xs">
            <NameBlock user={user} />
          </Stack>
        </Group>
        <Stack>
          <ButtonBlock user={user} myself={myself} />
        </Stack>
      </Group>
    );
  }
}

function AvatarBlock({ user, myself, friend }: Readonly<ProfileHeaderProps>) {
  return (
    <>
      <Avatar name={user.name || ""} image={user.image} />
      {getRoleBadge(user.role)}
      {myself && <ProfileBadge type="me" />}
      {friend && <ProfileBadge type="friend" />}
    </>
  );
}

function NameBlock({ user }: Readonly<Pick<ProfileHeaderProps, "user">>) {
  return (
    <>
      <ProfileUsername>{user.name}</ProfileUsername>
      {user.realName && <ProfileRealName>{user.realName}</ProfileRealName>}
      <ProfileAdditionalInformation
        place={user.place || undefined}
        bggName={user.bggName || undefined}
      />
    </>
  );
}

function ButtonBlock({ user, myself }: Readonly<ProfileHeaderProps>) {
  const { t } = useTranslation("profile");
  if (myself) {
    return (
      <>
        <ShareProfile profileId={user.id} />
        <LinkButton href="/app/profile/edit">{t("Edit")}</LinkButton>
      </>
    );
  } else {
    return <FriendRequestStatus userId={user.id} />;
  }
}

function getRoleBadge(role: Role) {
  switch (role) {
    case "ADMIN":
      return <ProfileBadge type="admin" />;
    case "FRIENDS_FAMILY":
      return <ProfileBadge type="family" />;
    case "PREMIUM":
      return <ProfileBadge type="premium" />;
    case "USER":
      return <ProfileBadge type="user" />;
  }
}
