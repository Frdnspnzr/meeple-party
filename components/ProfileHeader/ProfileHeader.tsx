"use client";

import { UserProfile } from "@/datatypes/userProfile";
import { useTranslation } from "@/i18n/client";
import { Group, Stack, Text, Title } from "@mantine/core";
import { Role } from "@prisma/client";
import Avatar from "../Avatar/Avatar";
import LinkButton from "../LinkButton/LinkButton";
import ProfileBadge from "../ProfileBadge/ProfileBadge";
import ShareProfile from "../ShareProfile/ShareProfile";
import styles from "./ProfileHeader.module.css";

export interface ProfileHeaderProps {
  user: UserProfile;
  myself?: boolean;
  friend?: boolean;
}

export default function ProfileHeader({
  user,
  myself = false,
  friend = false,
}: Readonly<ProfileHeaderProps>) {
  const { t } = useTranslation("profile");
  return (
    <Group justify="space-between">
      <Group gap="xl">
        <Stack gap="xs" align="center">
          <Avatar name={user.name || ""} image={user.image} />
          {getRoleBadge(user.role)}
          {myself && <ProfileBadge type="me" />}
          {friend && <ProfileBadge type="friend" />}
        </Stack>
        <Stack gap="xs">
          <Title order={2} className={styles.name}>
            {user.name}
          </Title>
          <Title order={3} className={styles.realName}>
            {user.realName}
          </Title>
          <Text className={styles.subline}>
            {t("Header.BggName", { name: user.bggName })}
          </Text>
        </Stack>
      </Group>
      {myself && (
        <Stack>
          <ShareProfile profileId={user.id} />
          <LinkButton href="/app/profile/edit">{t("Edit")}</LinkButton>
        </Stack>
      )}
    </Group>
  );
}

function getRoleBadge(role: Role) {
  switch (role) {
    case "ADMIN":
      return <ProfileBadge type="admin" />;
    case "FRIENDS_FAMILY":
      return <ProfileBadge type="family" />;
    case "PREMIUM":
      return <ProfileBadge type="premium" />;
    case "USER":
      return <ProfileBadge type="user" />;
  }
}
