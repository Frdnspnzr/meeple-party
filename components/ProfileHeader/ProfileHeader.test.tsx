import { render } from "@/utility/test";
import { Role } from "@prisma/client";
import ProfileHeader from "./ProfileHeader";

jest.mock("@/i18n/client");

describe("Profile header", () => {
  beforeAll(() => {
    Object.defineProperty(window, "matchMedia", {
      writable: true,
      value: jest.fn().mockImplementation((query) => ({
        matches: false,
        media: query,
        onchange: null,
        addListener: jest.fn(),
        removeListener: jest.fn(),
        addEventListener: jest.fn(),
        removeEventListener: jest.fn(),
        dispatchEvent: jest.fn(),
      })),
    });
  });
  it("matches snapshot for full user details", () => {
    const user = {
      name: "Frdnspnzr",
      realName: "Pascal Greilach",
      bggName: "Friedenspanzer",
      about: "Help, this thing is keeping me from playing games.",
      id: "abc123",
      place: "Kaiserslautern",
      role: Role.ADMIN,
      image: null,
    };
    const { container } = render(<ProfileHeader user={user} />);
    expect(container).toMatchSnapshot();
  });
  it("matches snapshot for minimal user details", () => {
    const user = {
      name: "Frdnspnzr",
      realName: null,
      bggName: null,
      about: null,
      id: "abc123",
      place: null,
      role: Role.USER,
      image: null,
    };
    const { container } = render(<ProfileHeader user={user} />);
    expect(container).toMatchSnapshot();
  });
  it("matches snapshot for friend profile", () => {
    const user = {
      name: "Frdnspnzr",
      realName: "Pascal Greilach",
      bggName: "Friedenspanzer",
      about: "Help, this thing is keeping me from playing games.",
      id: "abc123",
      place: "Kaiserslautern",
      role: Role.PREMIUM,
      image: null,
    };
    const { container } = render(<ProfileHeader user={user} friend />);
    expect(container).toMatchSnapshot();
  });
  it("matches snapshot for own profile", () => {
    const user = {
      name: "Frdnspnzr",
      realName: "Pascal Greilach",
      bggName: "Friedenspanzer",
      about: "Help, this thing is keeping me from playing games.",
      id: "abc123",
      place: "Kaiserslautern",
      role: Role.FRIENDS_FAMILY,
      image: null,
    };
    const { container } = render(<ProfileHeader user={user} myself />);
    expect(container).toMatchSnapshot();
  });
});
