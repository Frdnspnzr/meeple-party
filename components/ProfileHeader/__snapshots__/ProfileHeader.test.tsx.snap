// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile header matches snapshot for friend profile 1`] = `
<div>
  <style
    data-mantine-styles="classes"
  >
    @media (max-width: 35.99375em) {.mantine-visible-from-xs {display: none !important;}}@media (min-width: 36em) {.mantine-hidden-from-xs {display: none !important;}}@media (max-width: 47.99375em) {.mantine-visible-from-sm {display: none !important;}}@media (min-width: 48em) {.mantine-hidden-from-sm {display: none !important;}}@media (max-width: 61.99375em) {.mantine-visible-from-md {display: none !important;}}@media (min-width: 62em) {.mantine-hidden-from-md {display: none !important;}}@media (max-width: 74.99375em) {.mantine-visible-from-lg {display: none !important;}}@media (min-width: 75em) {.mantine-hidden-from-lg {display: none !important;}}@media (max-width: 87.99375em) {.mantine-visible-from-xl {display: none !important;}}@media (min-width: 88em) {.mantine-hidden-from-xl {display: none !important;}}
  </style>
  <div
    class="m-4081bf90 mantine-Group-root"
    style="--group-gap: var(--mantine-spacing-md); --group-align: center; --group-justify: space-between; --group-wrap: wrap;"
  >
    <div
      class="m-4081bf90 mantine-Group-root"
      style="--group-gap: var(--mantine-spacing-xl); --group-align: center; --group-justify: flex-start; --group-wrap: wrap;"
    >
      <div
        class="m-6d731127 mantine-Stack-root"
        style="--stack-gap: var(--mantine-spacing-xs); --stack-align: center; --stack-justify: flex-start;"
      >
        <div
          class="dummy"
          style="background-color: rgb(50, 82, 71);"
        >
          F
        </div>
        <div
          class="m-347db0ec mantine-Badge-root"
          style="--badge-radius: var(--mantine-radius-sm); --badge-bg: gold; --badge-color: var(--mantine-color-white); --badge-bd: calc(0.0625rem * var(--mantine-scale)) solid transparent;"
        >
          <span
            class="m-91fdda9b mantine-Badge-section"
            data-position="left"
          >
            <svg
              class="tabler-icon tabler-icon-star"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M12 17.75l-6.172 3.245l1.179 -6.873l-5 -4.867l6.9 -1l3.086 -6.253l3.086 6.253l6.9 1l-5 4.867l1.179 6.873z"
              />
            </svg>
          </span>
          <span
            class="m-5add502a mantine-Badge-label"
          >
            Badges.Premium
          </span>
        </div>
        <div
          class="m-347db0ec mantine-Badge-root"
          style="--badge-radius: var(--mantine-radius-sm); --badge-bg: var(--mantine-color-cyan-filled); --badge-color: var(--mantine-color-white); --badge-bd: calc(0.0625rem * var(--mantine-scale)) solid transparent;"
        >
          <span
            class="m-91fdda9b mantine-Badge-section"
            data-position="left"
          >
            <svg
              class="tabler-icon tabler-icon-friends"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M7 5m-2 0a2 2 0 1 0 4 0a2 2 0 1 0 -4 0"
              />
              <path
                d="M5 22v-5l-1 -1v-4a1 1 0 0 1 1 -1h4a1 1 0 0 1 1 1v4l-1 1v5"
              />
              <path
                d="M17 5m-2 0a2 2 0 1 0 4 0a2 2 0 1 0 -4 0"
              />
              <path
                d="M15 22v-4h-2l2 -6a1 1 0 0 1 1 -1h2a1 1 0 0 1 1 1l2 6h-2v4"
              />
            </svg>
          </span>
          <span
            class="m-5add502a mantine-Badge-label"
          >
            Badges.Friend
          </span>
        </div>
      </div>
      <div
        class="m-6d731127 mantine-Stack-root"
        style="--stack-gap: var(--mantine-spacing-xs); --stack-align: stretch; --stack-justify: flex-start;"
      >
        <h2
          class="name m-8a5d1357 mantine-Title-root"
          data-order="2"
          style="--title-fw: var(--mantine-h2-font-weight); --title-lh: var(--mantine-h2-line-height); --title-fz: var(--mantine-h2-font-size);"
        >
          Frdnspnzr
        </h2>
        <h3
          class="realName m-8a5d1357 mantine-Title-root"
          data-order="3"
          style="--title-fw: var(--mantine-h3-font-weight); --title-lh: var(--mantine-h3-line-height); --title-fz: var(--mantine-h3-font-size);"
        >
          Pascal Greilach
        </h3>
        <p
          class="mantine-focus-auto subline m-b6d8b162 mantine-Text-root"
        >
          Header.BggName
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`Profile header matches snapshot for full user details 1`] = `
<div>
  <style
    data-mantine-styles="classes"
  >
    @media (max-width: 35.99375em) {.mantine-visible-from-xs {display: none !important;}}@media (min-width: 36em) {.mantine-hidden-from-xs {display: none !important;}}@media (max-width: 47.99375em) {.mantine-visible-from-sm {display: none !important;}}@media (min-width: 48em) {.mantine-hidden-from-sm {display: none !important;}}@media (max-width: 61.99375em) {.mantine-visible-from-md {display: none !important;}}@media (min-width: 62em) {.mantine-hidden-from-md {display: none !important;}}@media (max-width: 74.99375em) {.mantine-visible-from-lg {display: none !important;}}@media (min-width: 75em) {.mantine-hidden-from-lg {display: none !important;}}@media (max-width: 87.99375em) {.mantine-visible-from-xl {display: none !important;}}@media (min-width: 88em) {.mantine-hidden-from-xl {display: none !important;}}
  </style>
  <div
    class="m-4081bf90 mantine-Group-root"
    style="--group-gap: var(--mantine-spacing-md); --group-align: center; --group-justify: space-between; --group-wrap: wrap;"
  >
    <div
      class="m-4081bf90 mantine-Group-root"
      style="--group-gap: var(--mantine-spacing-xl); --group-align: center; --group-justify: flex-start; --group-wrap: wrap;"
    >
      <div
        class="m-6d731127 mantine-Stack-root"
        style="--stack-gap: var(--mantine-spacing-xs); --stack-align: center; --stack-justify: flex-start;"
      >
        <div
          class="dummy"
          style="background-color: rgb(50, 82, 71);"
        >
          F
        </div>
        <div
          class="m-347db0ec mantine-Badge-root"
          style="--badge-radius: var(--mantine-radius-sm); --badge-bg: murple; --badge-color: var(--mantine-color-white); --badge-bd: calc(0.0625rem * var(--mantine-scale)) solid transparent;"
        >
          <span
            class="m-91fdda9b mantine-Badge-section"
            data-position="left"
          >
            <svg
              class="tabler-icon tabler-icon-code-circle-2"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M8.5 13.5l-1.5 -1.5l1.5 -1.5"
              />
              <path
                d="M15.5 10.5l1.5 1.5l-1.5 1.5"
              />
              <path
                d="M12 12m-9 0a9 9 0 1 0 18 0a9 9 0 1 0 -18 0"
              />
              <path
                d="M13 9.5l-2 5.5"
              />
            </svg>
          </span>
          <span
            class="m-5add502a mantine-Badge-label"
          >
            Badges.Admin
          </span>
        </div>
      </div>
      <div
        class="m-6d731127 mantine-Stack-root"
        style="--stack-gap: var(--mantine-spacing-xs); --stack-align: stretch; --stack-justify: flex-start;"
      >
        <h2
          class="name m-8a5d1357 mantine-Title-root"
          data-order="2"
          style="--title-fw: var(--mantine-h2-font-weight); --title-lh: var(--mantine-h2-line-height); --title-fz: var(--mantine-h2-font-size);"
        >
          Frdnspnzr
        </h2>
        <h3
          class="realName m-8a5d1357 mantine-Title-root"
          data-order="3"
          style="--title-fw: var(--mantine-h3-font-weight); --title-lh: var(--mantine-h3-line-height); --title-fz: var(--mantine-h3-font-size);"
        >
          Pascal Greilach
        </h3>
        <p
          class="mantine-focus-auto subline m-b6d8b162 mantine-Text-root"
        >
          Header.BggName
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`Profile header matches snapshot for minimal user details 1`] = `
<div>
  <style
    data-mantine-styles="classes"
  >
    @media (max-width: 35.99375em) {.mantine-visible-from-xs {display: none !important;}}@media (min-width: 36em) {.mantine-hidden-from-xs {display: none !important;}}@media (max-width: 47.99375em) {.mantine-visible-from-sm {display: none !important;}}@media (min-width: 48em) {.mantine-hidden-from-sm {display: none !important;}}@media (max-width: 61.99375em) {.mantine-visible-from-md {display: none !important;}}@media (min-width: 62em) {.mantine-hidden-from-md {display: none !important;}}@media (max-width: 74.99375em) {.mantine-visible-from-lg {display: none !important;}}@media (min-width: 75em) {.mantine-hidden-from-lg {display: none !important;}}@media (max-width: 87.99375em) {.mantine-visible-from-xl {display: none !important;}}@media (min-width: 88em) {.mantine-hidden-from-xl {display: none !important;}}
  </style>
  <div
    class="m-4081bf90 mantine-Group-root"
    style="--group-gap: var(--mantine-spacing-md); --group-align: center; --group-justify: space-between; --group-wrap: wrap;"
  >
    <div
      class="m-4081bf90 mantine-Group-root"
      style="--group-gap: var(--mantine-spacing-xl); --group-align: center; --group-justify: flex-start; --group-wrap: wrap;"
    >
      <div
        class="m-6d731127 mantine-Stack-root"
        style="--stack-gap: var(--mantine-spacing-xs); --stack-align: center; --stack-justify: flex-start;"
      >
        <div
          class="dummy"
          style="background-color: rgb(50, 82, 71);"
        >
          F
        </div>
        <div
          class="m-347db0ec mantine-Badge-root"
          style="--badge-radius: var(--mantine-radius-sm); --badge-bg: darkgray; --badge-color: var(--mantine-color-white); --badge-bd: calc(0.0625rem * var(--mantine-scale)) solid transparent;"
        >
          <span
            class="m-91fdda9b mantine-Badge-section"
            data-position="left"
          >
            <svg
              class="tabler-icon tabler-icon-user"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M8 7a4 4 0 1 0 8 0a4 4 0 0 0 -8 0"
              />
              <path
                d="M6 21v-2a4 4 0 0 1 4 -4h4a4 4 0 0 1 4 4v2"
              />
            </svg>
          </span>
          <span
            class="m-5add502a mantine-Badge-label"
          >
            Badges.User
          </span>
        </div>
      </div>
      <div
        class="m-6d731127 mantine-Stack-root"
        style="--stack-gap: var(--mantine-spacing-xs); --stack-align: stretch; --stack-justify: flex-start;"
      >
        <h2
          class="name m-8a5d1357 mantine-Title-root"
          data-order="2"
          style="--title-fw: var(--mantine-h2-font-weight); --title-lh: var(--mantine-h2-line-height); --title-fz: var(--mantine-h2-font-size);"
        >
          Frdnspnzr
        </h2>
        <h3
          class="realName m-8a5d1357 mantine-Title-root"
          data-order="3"
          style="--title-fw: var(--mantine-h3-font-weight); --title-lh: var(--mantine-h3-line-height); --title-fz: var(--mantine-h3-font-size);"
        />
        <p
          class="mantine-focus-auto subline m-b6d8b162 mantine-Text-root"
        >
          Header.BggName
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`Profile header matches snapshot for own profile 1`] = `
<div>
  <style
    data-mantine-styles="classes"
  >
    @media (max-width: 35.99375em) {.mantine-visible-from-xs {display: none !important;}}@media (min-width: 36em) {.mantine-hidden-from-xs {display: none !important;}}@media (max-width: 47.99375em) {.mantine-visible-from-sm {display: none !important;}}@media (min-width: 48em) {.mantine-hidden-from-sm {display: none !important;}}@media (max-width: 61.99375em) {.mantine-visible-from-md {display: none !important;}}@media (min-width: 62em) {.mantine-hidden-from-md {display: none !important;}}@media (max-width: 74.99375em) {.mantine-visible-from-lg {display: none !important;}}@media (min-width: 75em) {.mantine-hidden-from-lg {display: none !important;}}@media (max-width: 87.99375em) {.mantine-visible-from-xl {display: none !important;}}@media (min-width: 88em) {.mantine-hidden-from-xl {display: none !important;}}
  </style>
  <div
    class="m-4081bf90 mantine-Group-root"
    style="--group-gap: var(--mantine-spacing-md); --group-align: center; --group-justify: space-between; --group-wrap: wrap;"
  >
    <div
      class="m-4081bf90 mantine-Group-root"
      style="--group-gap: var(--mantine-spacing-xl); --group-align: center; --group-justify: flex-start; --group-wrap: wrap;"
    >
      <div
        class="m-6d731127 mantine-Stack-root"
        style="--stack-gap: var(--mantine-spacing-xs); --stack-align: center; --stack-justify: flex-start;"
      >
        <div
          class="dummy"
          style="background-color: rgb(50, 82, 71);"
        >
          F
        </div>
        <div
          class="m-347db0ec mantine-Badge-root"
          style="--badge-radius: var(--mantine-radius-sm); --badge-bg: mink; --badge-color: var(--mantine-color-white); --badge-bd: calc(0.0625rem * var(--mantine-scale)) solid transparent;"
        >
          <span
            class="m-91fdda9b mantine-Badge-section"
            data-position="left"
          >
            <svg
              class="tabler-icon tabler-icon-heart"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M19.5 12.572l-7.5 7.428l-7.5 -7.428a5 5 0 1 1 7.5 -6.566a5 5 0 1 1 7.5 6.572"
              />
            </svg>
          </span>
          <span
            class="m-5add502a mantine-Badge-label"
          >
            Badges.FriendsFamily
          </span>
        </div>
        <div
          class="m-347db0ec mantine-Badge-root"
          style="--badge-radius: var(--mantine-radius-sm); --badge-bg: var(--mantine-color-gray-filled); --badge-color: var(--mantine-color-white); --badge-bd: calc(0.0625rem * var(--mantine-scale)) solid transparent;"
        >
          <span
            class="m-91fdda9b mantine-Badge-section"
            data-position="left"
          >
            <svg
              class="tabler-icon tabler-icon-user-check"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M8 7a4 4 0 1 0 8 0a4 4 0 0 0 -8 0"
              />
              <path
                d="M6 21v-2a4 4 0 0 1 4 -4h4"
              />
              <path
                d="M15 19l2 2l4 -4"
              />
            </svg>
          </span>
          <span
            class="m-5add502a mantine-Badge-label"
          >
            Badges.You
          </span>
        </div>
      </div>
      <div
        class="m-6d731127 mantine-Stack-root"
        style="--stack-gap: var(--mantine-spacing-xs); --stack-align: stretch; --stack-justify: flex-start;"
      >
        <h2
          class="name m-8a5d1357 mantine-Title-root"
          data-order="2"
          style="--title-fw: var(--mantine-h2-font-weight); --title-lh: var(--mantine-h2-line-height); --title-fz: var(--mantine-h2-font-size);"
        >
          Frdnspnzr
        </h2>
        <h3
          class="realName m-8a5d1357 mantine-Title-root"
          data-order="3"
          style="--title-fw: var(--mantine-h3-font-weight); --title-lh: var(--mantine-h3-line-height); --title-fz: var(--mantine-h3-font-size);"
        >
          Pascal Greilach
        </h3>
        <p
          class="mantine-focus-auto subline m-b6d8b162 mantine-Text-root"
        >
          Header.BggName
        </p>
      </div>
    </div>
    <div
      class="m-6d731127 mantine-Stack-root"
      style="--stack-gap: var(--mantine-spacing-md); --stack-align: stretch; --stack-justify: flex-start;"
    >
      <div
        class="m-4081bf90 mantine-Group-root"
        style="--group-gap: var(--mantine-spacing-md); --group-align: center; --group-justify: flex-start; --group-wrap: wrap;"
      >
        <button
          class="mantine-focus-auto mantine-active m-77c9d27d mantine-Button-root m-87cf2631 mantine-UnstyledButton-root"
          data-variant="filled"
          data-with-left-section="true"
          style="--button-bg: var(--mantine-color-blue-filled); --button-hover: var(--mantine-color-blue-filled-hover); --button-color: var(--mantine-color-white); --button-bd: calc(0.0625rem * var(--mantine-scale)) solid transparent;"
          type="button"
        >
          <span
            class="m-80f1301b mantine-Button-inner"
          >
            <span
              class="m-a74036a mantine-Button-section"
              data-position="left"
            >
              <svg
                class="tabler-icon tabler-icon-share-2"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M8 9h-1a2 2 0 0 0 -2 2v8a2 2 0 0 0 2 2h10a2 2 0 0 0 2 -2v-8a2 2 0 0 0 -2 -2h-1"
                />
                <path
                  d="M12 14v-11"
                />
                <path
                  d="M9 6l3 -3l3 3"
                />
              </svg>
            </span>
            <span
              class="m-811560b9 mantine-Button-label"
            >
              Share
            </span>
          </span>
        </button>
      </div>
      <a
        class="mantine-focus-auto mantine-active m-77c9d27d mantine-Button-root m-87cf2631 mantine-UnstyledButton-root"
        href="/app/profile/edit"
      >
        <span
          class="m-80f1301b mantine-Button-inner"
        >
          <span
            class="m-811560b9 mantine-Button-label"
          >
            Edit
          </span>
        </span>
      </a>
    </div>
  </div>
</div>
`;
